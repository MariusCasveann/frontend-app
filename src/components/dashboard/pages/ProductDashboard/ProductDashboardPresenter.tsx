import * as React from 'react';
import { CardLayout } from '../../../common/molecules/CardLayout/CardLayout';
import AccumulatedProductCostPresenter from '../../molecules/AccumulatedProductCost/AccumulatedProductCostPresenter';
import CpuUtilizationPresenter from '../../molecules/CpuUtilization/CpuUtilizationPresenter';
import '../common/Dashboard.css';
import { ProductFiltersPresenter } from './ProductFiltersPresenter';
import { useState } from 'react';
import { QueryAllProducts_allProducts } from '../../atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import { getCurrentDate, getYesterdayDate } from '../../../../utils/fiscalYear';
import { FiltersContext } from '../../atoms/Filters/FiltersContext';
import DeploymentFrequencyPerProductPresenter from '../../molecules/DeploymentFrequencyPerProduct/DeploymentFrequencyPerProductPresenter';
import CostPerProductPresenter from '../../molecules/CostPerProduct/CostPerProductPresenter';

export default () => {
    const [selectedProductState, setSelectedProduct] = useState<QueryAllProducts_allProducts | undefined>(undefined);
    const [fromDateState, setFromDate] = useState<string>(getYesterdayDate());
    const [toDateState, setToDate] = useState<string>(getCurrentDate());
    const [financialYearState, setFinancialYear] = useState<string | null>(null);

    return (
        <FiltersContext.Provider
            value={{
                fromDate: fromDateState,
                toDate: toDateState,
                setToDate,
                setFromDate,
                financialYear: financialYearState,
                setFinancialYear
            }}
        >
            <div className="dashboard-wrapper">
                <CardLayout width={100} className="dashboard-card-layout">
                    <ProductFiltersPresenter
                        selectedProduct={selectedProductState}
                        setSelectedProduct={setSelectedProduct}
                    />
                    <AccumulatedProductCostPresenter selectedProduct={selectedProductState} />
                    <DeploymentFrequencyPerProductPresenter selectedProduct={selectedProductState} />
                    <CardLayout width={100}>
                        <CostPerProductPresenter selectedProduct={selectedProductState} />
                        <CpuUtilizationPresenter selectedProduct={selectedProductState} />
                    </CardLayout>
                </CardLayout>
            </div>
        </FiltersContext.Provider>
    );
};
