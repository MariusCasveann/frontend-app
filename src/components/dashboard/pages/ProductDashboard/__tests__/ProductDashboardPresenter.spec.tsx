import * as React from 'react';
jest.mock('../../../../common/molecules/CardLayout/CardLayout', () => ({ Card: 'Card', CardLayout: 'CardLayout' }));
jest.mock(
    '../../../molecules/AccumulatedProductCost/AccumulatedProductCostPresenter',
    () => 'AccumulatedProductCostPresenter'
);
jest.mock(
    '../../../molecules/DeploymentFrequencyPerProduct/DeploymentFrequencyPerProductPresenter',
    () => 'DeploymentFrequencyPerProductPresenter'
);
jest.mock('../../../molecules/CostPerProduct/CostPerProductPresenter', () => 'CostPerProductPresenter');
jest.mock('../../../molecules/CpuUtilization/CpuUtilizationPresenter', () => 'CpuUtilizationPresenter');
jest.mock('../ProductFiltersPresenter', () => ({ ProductFiltersPresenter: 'ProductFiltersPresenter' }));

import ProductDashboardPresenter from '../ProductDashboardPresenter';
import renderer from 'react-test-renderer';

describe(' ProductDashboardPresenter', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<ProductDashboardPresenter />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
