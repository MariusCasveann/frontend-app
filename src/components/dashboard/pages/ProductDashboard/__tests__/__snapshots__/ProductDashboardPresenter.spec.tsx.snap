// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` ProductDashboardPresenter renders correctly 1`] = `
<div
  className="dashboard-wrapper"
>
  <CardLayout
    className="dashboard-card-layout"
    width={100}
  >
    <ProductFiltersPresenter
      setSelectedProduct={[Function]}
    />
    <AccumulatedProductCostPresenter />
    <DeploymentFrequencyPerProductPresenter />
    <CardLayout
      width={100}
    >
      <CostPerProductPresenter />
      <CpuUtilizationPresenter />
    </CardLayout>
  </CardLayout>
</div>
`;
